src/Monitor/Monitor/ScheduledQueryRules/NewScheduledQueryRuleCommand.cs

```diff
@@ -0,0 +1,128 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Insights.OutputClasses;
+using Microsoft.Azure.Management.Monitor.Models;
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+
+namespace Microsoft.Azure.Commands.Insights.ScheduledQueryRules
+{
+    /// <summary>
+    /// Create a ScheduledQueryRule Source object
+    /// </summary>
+    [Cmdlet(VerbsCommon.New, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "ScheduledQueryRule", SupportsShouldProcess = true), OutputType(typeof(PSScheduledQueryRuleResource))]
+    public class NewScheduledQueryRuleCommand : ManagementCmdletBase
+    {
+
+        #region Cmdlet parameters
+
+        //
+        // Summary:
+        //     Gets or sets source (Query, DataSourceId, etc.) for rule.
+        [Parameter(Mandatory = true, HelpMessage = "The scheduled query rule source")]
+        [ValidateNotNullOrEmpty]
+        public PSScheduledQueryRuleSource Source { get; set; }
+
+        //
+        // Summary:
+        //     Gets or sets schedule (Frequnecy, Time Window) for rule.
+        [Parameter(Mandatory = true, HelpMessage = "The scheduled query rule schedule")]
+        [ValidateNotNullOrEmpty]
+        public PSScheduledQueryRuleSchedule Schedule { get; set; }
+
+        //
+        // Summary:
+        //     Gets or sets action needs to be taken on rule execution.
+        [Parameter(Mandatory = true, HelpMessage = "The scheduled query rule Alerting Action")]
+        [ValidateNotNullOrEmpty]
+        public PSScheduledQueryRuleAlertingAction Action { get; set; }
+
+        //
+        // Summary:
+        //     Region where alert is to be created
+        [Parameter(Mandatory = true, HelpMessage = "The location for this alert")]
+        [ValidateNotNullOrEmpty]
+        public string Location { get; set; }
```

Comment:
@shpimpal this should have the [`LocationCompleter`](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/design-guidelines/parameter-best-practices.md#location-completer) attribute to allow users to tab through the valid locations

Reply:
will add it.